### PR TITLE
Move mail settings into a modal

### LINKED ISSUE #6896: App settings should be a modal
### Is your feature request related to a problem? Please describe.

Currently app settings are located in the lower left corner. This makes the view very cramped and off sight, and requires lots of scrolling on a tiny space. It also makes the settings button move up and melt with the list of inboxes and folder, without any visual separation.

image

### Describe the solution you'd like

Like Talk, settings should be displayed in a modal view.

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by GretaD**
- [X] Move settings into a modal via AppSettingsDialogs dialog
- [x] Polish the new modal
- [x] Move the global Trusted senders from account settings modals to the app settings modal.

fixes #6896

### CODE REVIEW

**Comment by GretaD** [comment]
the view so far. Feel free to suggest different titles for the sections.

[Screencast from 07.12.2023 12:14:25.webm](https://github.com/nextcloud/mail/assets/12728974/953fb3bf-6189-4378-8623-6a03d9246195)

File: src/components/Navigation.vue [hunk 1]
```diff
@@ -242,5 +251,13 @@ to {
 		}
 	}
 }
+.mail-settings {
+	padding: calc(var(--default-grid-baseline, 4px) * 2);
+}
+.mail-settings-button {
+	width: 100% !important;
+	justify-content: start !important;
+
```
Comment (ChristophWurst) [inline src/components/Navigation.vue:260]: :ghost:

**Review by hamza221 — APPROVED** [review]
LGTM , Tested ✅

File: src/components/AppSettingsMenu.vue [hunk 2]
```diff
@@ -1,158 +1,186 @@
...
-		<p class="mailvelope-section">
-			{{ t('mail', 'Looking for a way to encrypt your emails?') }}
-		</p>
-		<a
-			href="https://www.mailvelope.com/"
-			target="_blank"
-			rel="noopener noreferrer">
-			{{ t('mail', 'Install Mailvelope browser extension here') }}
-		</a>
+	<div class="appppp-settings">
```
[the remark this answers is not in the document]
Comment (GretaD) [reply src/components/AppSettingsMenu.vue:2]: yes :face_in_clouds:

[the remark this answers is not in the document]
Comment (ChristophWurst) [reply src/components/AppSettingsMenu.vue:2]: https://www.youtube.com/watch?v=luLX6gCbPyw

**Review by st3iny — COMMENTED** [review]
Eslint is not happy. Running `npm run lint:fix` should fix all issues.

**Comment by st3iny** [comment]
I tested it briefly and it works well.

File: src/components/AppSettingsMenu.vue [hunk 3]
```diff
@@ -1,147 +1,177 @@
...
+				<TrustedSenders />
+			</NcAppSettingsSection>
+			<NcAppSettingsSection id="gavatar-settings" :title="t('mail', 'Gavatar settings')">
+				<p v-if="loadingAvatarSettings" class="app-settings avatar-settings">
+					<IconLoading :size="20" />
+					{{ t('mail', 'Use Gravatar and favicon avatars') }}
+				</p>
+				<p v-else class="app-settings">
+					<input
+						id="gravatar-enabled"
```
Comment (ChristophWurst) [inline src/components/AppSettingsMenu.vue:73]:
```suggestion
					<input id="gravatar-enabled"
```

File: src/components/AppSettingsMenu.vue [hunk 4]
```diff
@@ -1,147 +1,177 @@
...
+						{{ t('mail', 'Oldest') }}
+					</CheckboxRadioSwitch>
+				</div>
+			</NcAppSettingsSection>
+			<NcAppSettingsSection id="mailvelope-settings" :title="t('mail', 'Mailvelope')">
+				<p class="mailvelope-section">
+					{{ t('mail', 'Looking for a way to encrypt your emails?') }}
+				</p>
+				<a
+					href="https://www.mailvelope.com/"
```
Comment (ChristophWurst) [inline src/components/AppSettingsMenu.vue:160]:
```suggestion
				<a href="https://www.mailvelope.com/"
```

**Review by ChristophWurst — COMMENTED** [review]
eslint

**Comment by GretaD** [comment]
> Eslint is not happy. Running `npm run lint:fix` should fix all issues.

this never works for me, thats why i did it manually :(

**Pull request merged**